src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -34,6 +35,16 @@ type ClusterSpec struct {
 	Resources corev1.ResourceRequirements `json:"resources"`
 	// Configuration represent redpanda specific configuration
 	Configuration RedpandaConfig `json:"configuration,omitempty"`
+	// Storage spec for cluster
+	Storage StorageSpec `json:"storage,omitempty"`
+}
+
+// StorageSpec defines the storage specification of the Cluster
+type StorageSpec struct {
```

Comment:
should we add some validation for this into webhook?

Reply:
There are two candidate quantities for validation, the capacity and the storage class. Are you thinking of
- a lower/upper bound on the capacity?
- validating that the storage class exists (within the same namespace)?

Reply:
we could do capacity > 0 for start, I think we can't validate any upper bounds if we don't know what node will it land, right?

as to the storage class, can we even validate that exists?

I am fine with this being a follow-up if we create an issue. I mean the > 0 validation is probably trivial :)

Reply:
storage class isn't namespaced:

```
NAME                              SHORTNAMES   APIVERSION                        NAMESPACED   KIND
storageclasses               sc                        storage.k8s.io/v1                false                   StorageClass
```

Reply:
> as to the storage class, can we even validate that exists?
I guess not in the validator (but in the controller instead).